### PR TITLE
TRITON-2079 CNAPI should provide information regarding Platform OS

### PR DESCRIPTION
**Pull request by kusor**
Like the following, including more info for linux PIs:

```
[root@headnode (coal) ~]# sdc-cnapi /platforms|json -H
{
  "20190523T221915Z": {},
  "20190613T185529Z": {},
  "20190710T121115Z": {},
  "20191218T190857Z": {},
  "20200220T122949Z": {
    "latest": true
  }
}
[root@headnode (coal) ~]# sdc-cnapi /platforms?os=false|json -H
{
  "20190523T221915Z": {},
  "20190613T185529Z": {},
  "20190710T121115Z": {},
  "20191218T190857Z": {},
  "20200220T122949Z": {
    "latest": true
  }
}
[root@headnode (coal) ~]# sdc-cnapi /platforms?os=true|json -H
{
  "20190523T221915Z": {
    "os": "smartos"
  },
  "20190613T185529Z": {
    "os": "smartos"
  },
  "20190710T121115Z": {
    "os": "smartos"
  },
  "20191218T190857Z": {
    "os": "smartos"
  },
  "20200220T122949Z": {
    "os": "smartos",
    "latest": true
  }
}
```

### CODE REVIEW

File: lib/models/platform.js [hunk 1]
```diff
@@ -99,15 +117,35 @@ ModelPlatform.list = function (params, callback) {
...
                 return RE.test(img);
             }).sort();
-            var latest = imgs.pop();
-            if (latest) {
-                platformImages[latest].latest = true;
+
+            if (params.os) {
+                var linuxImgs = imgs.filter(function isLinux(p) {
+                    return (platformImages[p].os &&
+                        platformImages[p].os === 'linux');
```
Comment (twhiteman) [inline lib/models/platform.js]:
```suggestion
                    return (platformImages[p].os === 'linux');
```

File: lib/models/platform.js [hunk 2]
```diff
@@ -99,15 +117,35 @@ ModelPlatform.list = function (params, callback) {
...
+                });
+                if (linuxImgs.length) {
+                    var latestLinux = linuxImgs.pop();
+                    platformImages[latestLinux].latest = true;
+                }
+                var smartosImgs = imgs.filter(function isSmartos(p) {
+                    return (platformImages[p].os &&
+                        platformImages[p].os === 'smartos');
```
Comment (twhiteman) [inline lib/models/platform.js]:
```suggestion
                    return (platformImages[p].os === 'smartos');
```

File: lib/models/platform.js (hunk 1 above)
Comment (kusor) [reply lib/models/platform.js]: Done!

File: lib/models/platform.js (hunk 2 above)
Comment (kusor) [reply lib/models/platform.js]: Done!

**Comment by twhiteman** [comment]
> I wonder if this 'os' query would be better to be passed in though an 'extras' field - i.e. to keep it working the same way as CNAPI servers currently works:
> 
> ```
> /platforms?extras=os
> /servers?extras=sysinfo
> ```

@kusor what's your thoughts on this ^ ?

**Comment by kusor** [comment]
> I wonder if this 'os' query would be better to be passed in though an 'extras' field - i.e. to keep it working the same way as CNAPI servers currently works:
> 
> ```
> /platforms?extras=os
> /servers?extras=sysinfo
> ```

I don't have a strong opinion on this. I'd say that for platforms are so few possible options that I would just go for `?os=true` ... unless we can figure out that it could really become something more complex like servers listing?

**Comment by twhiteman** [comment]
Okay, I'll leave it for your judgement on this (i.e. feel free to leave it as is).

We also need some doc changes (for the os query param) and a package.json bump.

**Comment by kusor** [comment]
> Okay, I'll leave it for your judgement on this (i.e. feel free to leave it as is).
> 

I'll keep it as is for now, yeah.

> We also need some doc changes (for the os query param) and a package.json bump.

Done sir!

**Pull request merged**